src/air/public_memory.cairo

```diff
@@ -43,6 +43,48 @@ impl PageImpl of PageTrait {
             i += 1;
         }
     }
+
+    fn extract_range(self: @Page, addr: felt252, len: felt252) -> Span<felt252> {
+        let mut arr = ArrayTrait::new();
+        let mut i = 0;
+
+        loop {
+            if i == len {
+                break arr.span();
+            }
+
+            let current = *self.at((addr + i).try_into().unwrap());
+
+            // TODO is this needed? If not we can just use slice directly 
+            assert(current.address == addr + i, 'Invalid address');
+            arr.append(current.value);
+            i += 1;
+        }
+    }
+
+    fn verify_stack(
+        self: @Page,
+        start_ap: felt252,
+        segment_address: felt252,
+        builtins: Span<felt252>,
+        memory_index: felt252
+    ) {
+        let mut i = 0;
+
+        // TODO size of SegmentInfo
+        let size = 2;
```

Comment:
can we add constant SegmentInfoSize?

Reply:
Yes I want to do this but this will also depend on the type of segment I think (program, execution, output). I want to get back into this when testing, and have some real values to work with